nixos/modules/services/networking/mbim-network.nix

```diff
@@ -0,0 +1,127 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.mbim-network;
+
+  # Transforms "/dev/cdc-wdm0" into "mbim-network-dev-cdc-wdm0" for file and
+  # systemd unit names.
+  cleanName = name: "mbim-network${replaceStrings [ "/" ] [ "-" ] name}";
+
+  makeSystemdUnit = dcfg: name:
+    let
+      # Config file format defined in 'man 1 mbim-network'.
+      configFile = with dcfg;
+        pkgs.writeText "${cleanName name}.conf" ''
+          APN=${apn}
+          PROXY=${if cfg.proxy then "yes" else "no"}
+          ${optionalString (apnUser != "") "APN_USER=${apnUser}"}
+          ${optionalString (apnPass != "") "APN_PASS=${apnPass}"}
```

Comment:
This persist the APN Password in the nix store - which is probably fine for public APNs, as they're broadly known and in `mobile-broadband-provider-info`.

Probably enough to add a line into the module option description noting it ends up there.

Reply:
Done!